### PR TITLE
[assertj] Add assertions for BundleEvent, BundleReference, BundleContext

### PR DESCRIPTION
**Pull request by kriegfrj**

### CODE REVIEW

File: org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundlecontext/BundleContextSoftAssertions.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.bundlecontext;
+
+import org.assertj.core.api.SoftAssertionsProvider;
+import org.osgi.framework.BundleContext;
+
+public interface BundleContextSoftAssertions extends SoftAssertionsProvider {
```
Comment (bjhargrave) [inline org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundlecontext/BundleContextSoftAssertions.java:22]: Shouldn't we name these soft assertion interfaces to end in `SoftAssertionsProvider` like the others? Seems like nice consistency.

File: org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventSoftAssertions.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.bundleevent;
+
+import org.assertj.core.api.SoftAssertionsProvider;
+import org.osgi.framework.BundleEvent;
+
+public interface BundleEventSoftAssertions extends SoftAssertionsProvider {
```
Comment (bjhargrave) [inline org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundleevent/BundleEventSoftAssertions.java:22]: Name question here too.

File: org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundlereference/BundleReferenceSoftAssertions.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.osgi.test.assertj.bundlereference;
+
+import org.assertj.core.api.SoftAssertionsProvider;
+import org.osgi.framework.BundleReference;
+
+public interface BundleReferenceSoftAssertions extends SoftAssertionsProvider {
```
Comment (bjhargrave) [inline org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundlereference/BundleReferenceSoftAssertions.java:22]: Name question here too.

File: org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundlecontext/BundleContextSoftAssertions.java (hunk 1 above)
Comment (kriegfrj) [reply org.osgi.test.assertj/src/main/java/org/osgi/test/assertj/bundlecontext/BundleContextSoftAssertions.java]: Absolutely!

**Pull request merged**